### PR TITLE
Added Certify VEX statement schema and testing backend

### PR DESCRIPTION
**Pull request by pxp928**
CertifyVEXStatement:

- Description: attestation that the package/artifact has a VEX with a justification of its relationship with a CVE or GHSA

- Subject:      pkg / artifact 
- Object:        about ( CVE/GHSA )
- Properties:  justification ( string )
                      knownSince ( timestamp )
                      source ( string )
                      collector ( string )

```
fragment allCertifyVEXStatement on CertifyVEXStatement {
  subject {
    __typename
    ... on Package {
      type
      namespaces {
        namespace
        names {
          name
          versions {
            version
            qualifiers {
              key
              value
            }
            subpath
          }
        }
      }
    }
    ... on Artifact {
      algorithm
      digest
    }
  }
  vulnerability {
    __typename
    ... on CVE {
      year
      cveId {
        id
      }
    }
    ... on GHSA {
      ghsaId {
        id
      }
    }
  }
  justification
  knownSince
  origin
  collector
}

query Q1 {
  CertifyVEXStatement(certifyVEXStatementSpec: {}) {
    ...allCertifyVEXStatement
  }
}

query Q2 {
  CertifyVEXStatement(certifyVEXStatementSpec: {origin: "testing backend"}) {
    ...allCertifyVEXStatement
  }
}

query Q3 {
  CertifyVEXStatement(certifyVEXStatementSpec: {package: {name: "openssl"}}) {
    ...allCertifyVEXStatement
  }
}


query Q4 {
  CertifyVEXStatement(certifyVEXStatementSpec: {artifact: {algorithm: "sha1", digest: "5a787865sd676dacb0142afa0b83029cd7befd9"}}) {
    ...allCertifyVEXStatement
  }
}


query Q5 {
  CertifyVEXStatement(certifyVEXStatementSpec: {cve: {cveId: "CVE-2019-13110"}}) {
    ...allCertifyVEXStatement
  }
}

query Q6 {
  CertifyVEXStatement(certifyVEXStatementSpec: {ghsa: {ghsaId: "GHSA-h45f-rjvw-2rv2"}}) {
    ...allCertifyVEXStatement
  }
}
```

### CODE REVIEW

**Comment by pxp928** [comment]
Note: This will rebase once https://github.com/guacsec/guac/pull/467 is merged!

File: pkg/assembler/graphql/schema/certifyVEXStatement.graphql [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  justification: String
+  knownSince: String
+  origin: String
+  collector: String
+}
+
+"""
+PkgArtObject is a union of Package and Artifact. Any of these objects can be specified
+"""
+union PkgArtObject = Package | Artifact
```
Comment (lumjjb) [inline pkg/assembler/graphql/schema/certifyVEXStatement.graphql:57]: should we name this something like `SoftwareObject`?

File: pkg/assembler/graphql/schema/isVulnerability.graphql [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  ghsa: GHSASpec
+  justification: String
+  origin: String
+  collector: String
+}
+
+"""
+CveGhsaObject is a union of CVE and GHSA.
+"""
+union CveGhsaObject = CVE | GHSA
```
Comment (lumjjb) [inline pkg/assembler/graphql/schema/isVulnerability.graphql:53]: should we name this `VulnObject`?

**Review by lumjjb — APPROVED** [review]

File: pkg/assembler/graphql/schema/isVulnerability.graphql (hunk 2 above)
Comment (pxp928) [reply pkg/assembler/graphql/schema/isVulnerability.graphql:53]: `certifyVuln` has another union type that consists of CVE, OSV, and GHSA. So this was done to keep them separate

File: pkg/assembler/graphql/schema/certifyVEXStatement.graphql (hunk 1 above)
Comment (pxp928) [reply pkg/assembler/graphql/schema/certifyVEXStatement.graphql:57]: Similarly, there are other union types that consist of pkg, artifact, and source or pkg and source. So this was done to keep them apart from one other.

**Comment by pxp928** [comment]
i mean instead of having individual bools, we have one bool

matched := true

and then each check it goes through it does
match &= false if it does not pass the check

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**